tools/generate-helpers/pg-table-bindings/store.go.tpl

```diff
@@ -893,12 +897,32 @@ func (s *storeImpl) DeleteMany(ctx context.Context, ids []{{$singlePK.Type}}) er
     {{- end }}
     {{- end }}{{/* if not $inMigration */}}
 
-    q := search.ConjunctionQuery(
-    sacQueryFilter,
-        search.NewQueryBuilder().AddDocIDs(ids...).ProtoQuery(),
-    )
+    // Batch the deletes
+    batchSize := deleteBatchSize
+    for {
+        if len(ids) == 0 {
+            break
+        }
+
+        if len(ids) < batchSize {
+            batchSize = len(ids)
+        }
 
-    return postgres.RunDeleteRequestForSchema(ctx, schema, q, s.db)
+        idBatch := ids[0:batchSize]
+        q := search.ConjunctionQuery(
+        sacQueryFilter,
+            search.NewQueryBuilder().AddDocIDs(idBatch...).ProtoQuery(),
+        )
+
+        if err := postgres.RunDeleteRequestForSchema(ctx, schema, q, s.db); err != nil {
+            return err
```

Comment:
Since the `DeleteMany` operation is split into individual transactions, it would be nice to log as well as return in error how many entries were actually deleted.